chessington/engine/pieces.py

```diff
@@ -33,14 +36,45 @@ class Pawn(Piece):
     """
     A class representing a chess pawn.
     """
+
     def get_available_moves(self, board) -> List[Square]:
+        available_moves = []
         current_square = board.find_piece(self)
+        possible_squares = self.get_possible_squares(board, current_square)
         if self.player == Player.BLACK:
-            square_in_front = Square.at(current_square.row - 1, current_square.col)
-            return [square_in_front]
+            starting_row = 6
+            starting_color = Player.BLACK
+        else:
+            starting_row = 1
+            starting_color = Player.WHITE
+
+        if board.is_inside_bounds(possible_squares[0]) is True and board.get_piece(possible_squares[0]) is None:
+            available_moves.append(possible_squares[0])
+            if board.is_inside_bounds(possible_squares[1]) is True and board.get_piece(possible_squares[1]) is None:
+                if self.player == starting_color and current_square.row == starting_row:
+                    available_moves.append(possible_squares[1])
+
+        for i in (2,3):
+            if board.is_inside_bounds(possible_squares[i]) is True and board.get_piece(possible_squares[i]) is not None:
+                if board.get_piece(possible_squares[i]).player != self.player:
+                    available_moves.append(possible_squares[i])
+
+        return available_moves
+
+    def get_possible_squares(self, board, current_square):
+        possible_squares = []
+        if board.get_piece(current_square).player == Player.BLACK:
+            possible_squares.append(Square.at(current_square.row - 1, current_square.col))  # 1 up -> 0
+            possible_squares.append(Square.at(current_square.row - 2, current_square.col))  # 2 up -> 1
+            possible_squares.append(Square.at(current_square.row - 1, current_square.col - 1))  # take left -> 2
+            possible_squares.append(Square.at(current_square.row - 1, current_square.col + 1))  # take right -> 3
         else:
-            square_in_front = Square.at(current_square.row + 1, current_square.col)
-            return [square_in_front]
+            possible_squares.append(Square.at(current_square.row + 1, current_square.col))
+            possible_squares.append(Square.at(current_square.row + 2, current_square.col))
+            possible_squares.append(Square.at(current_square.row + 1, current_square.col - 1))
+            possible_squares.append(Square.at(current_square.row + 1, current_square.col + 1))
```

Comment:
There is something very similar about these branches.

Hint: look at `current_square.row`